services/src/main/java/org/keycloak/protocol/oid4vc/issuance/OID4VCIssuerWellKnownProvider.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.protocol.oid4vc.issuance;
+
+import jakarta.ws.rs.core.UriInfo;
+import org.keycloak.models.KeycloakContext;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.protocol.oid4vc.OID4VCClientRegistrationProvider;
+import org.keycloak.protocol.oid4vc.OID4VCLoginProtocolFactory;
+import org.keycloak.protocol.oid4vc.issuance.signing.VCSigningServiceProviderFactory;
+import org.keycloak.protocol.oid4vc.issuance.signing.VerifiableCredentialsSigningService;
+import org.keycloak.protocol.oid4vc.model.CredentialIssuer;
+import org.keycloak.protocol.oid4vc.model.Format;
+import org.keycloak.protocol.oid4vc.model.OID4VCClient;
+import org.keycloak.protocol.oid4vc.model.SupportedCredential;
+import org.keycloak.services.Urls;
+import org.keycloak.urls.UrlType;
+import org.keycloak.wellknown.WellKnownProvider;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * {@link  WellKnownProvider} implementation to provide the .well-known/openid-credential-issuer endpoint, offering
+ * the Credential Issuer Metadata as defined by the OID4VCI protocol
+ * {@see https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#section-10.2.2}
+ *
+ * @author <a href="https://github.com/wistefan">Stefan Wiedemann</a>
+ */
+public class OID4VCIssuerWellKnownProvider implements WellKnownProvider {
+
+    private final KeycloakSession keycloakSession;
+
+    public OID4VCIssuerWellKnownProvider(KeycloakSession keycloakSession) {
+        this.keycloakSession = keycloakSession;
+    }
+
+    @Override
+    public void close() {
+        // no-op
+    }
+
+    @Override
+    public Object getConfig() {
+        return new CredentialIssuer()
+                .setCredentialIssuer(getIssuer(keycloakSession.getContext()))
+                .setCredentialEndpoint(getCredentialsEndpoint(keycloakSession.getContext()))
+                .setCredentialsSupported(getSupportedCredentials(keycloakSession))
+                .setAuthorizationServers(List.of(getIssuer(keycloakSession.getContext())));
```

Comment:
The VCI specifications states:

> If this parameter is omitted, the entity providing the Credential Issuer is also acting as the Authorization Server, i.e., the Credential Issuer's identifier is used to obtain the Authorization Server metadata. The actual OAuth 2.0 Authorization Server metadata is obtained from the oauth-authorization-server well-known location as defined in Section 3 of [[RFC8414](https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#RFC8414)].

To my understanding this means that  KC could skip setting the authorization servers list. Even if it is set, I see no harm though.

Reply:
I agree in general. According to the spec, it would be optional and can be empty in our case. However, there are a number of wallets out there, that handle it as mandatory and fail without. Especially the once listed under [the EBSI-conformance list](https://ec.europa.eu/digital-building-blocks/sites/display/EBSI/Conformant+wallets) are susceptible to that, since the conformance test provides the authorization_servers. Thus, I decided to set it, since that is still valid with the spec.

Reply:
@wistefan I think draft 13 will "break" many wallet implementations :smiley: 
This field used do be a single value (not a list).  

Anyway. No problem if set.